doc/modules/gaussian_process.rst

```diff
@@ -484,6 +484,8 @@ Note that magic methods ``__add__``, ``__mul___`` and ``__pow__`` are
 overridden on the Kernel objects, so one can use e.g. ``RBF() + RBF()`` as
 a shortcut for ``Sum(RBF(), RBF())``.
 
+Note: CompoundKernel cannot be used in `GaussianProcessClassifier`
+
```

Comment:
I do not think we need to place this in the user guide. I think it's better placed in the parameter description:

https://github.com/scikit-learn/scikit-learn/blob/1d1aadd0711b87d2a11c80aad15df6f8cf156712/sklearn/gaussian_process/_gpc.py#L52-L55

this is the place most will be looking at when trying to construct their own kernels.

Reply:
I will update that instead